File: Halstead_java_example/src/Calculator.java

public class Calculator {

<<<<<<< Updated upstream
    // yeeeaaa im a comment
=======
    //adds to positive numbers
    //faulty inputs should return -1
>>>>>>> Stashed changes
    public int add(int a, int b) {
        if (a < 0 || b < 0) {
            return -1;
        }

        if (a + b >= Integer.MAX_VALUE || a + b < 0) {
            return -1;
        }
        return a + b;
    }

<<<<<<< Updated upstream
=======
    // substract to positive numbers
    //faulty inputs should return -1
>>>>>>> Stashed changes
    public int sub(int c, int d) {
        if (c < 0 || d < 0) {
            return -1;
        }
        return c - d;
    }

<<<<<<< Updated upstream
=======
    // devide one/two if they are positive
    //faulty inputs should return -1
>>>>>>> Stashed changes
    public int divide(int one, int two) {
        if (two == 0) {
            return -1;
        }
        return one / two;
    }
<<<<<<< Updated upstream

    public int sum(int first, int second) {
        if(first == Integer.MAX_VALUE || second == Integer.MAX_VALUE){
            return -1;
        }
        if (first * second >= Integer.MAX_VALUE) {
            return -1;
        }
        return first * second;
=======
    // sum two positive integers 
    //faulty inputs should return -1
    public int sum(int first, int second){
        return 0;
>>>>>>> Stashed changes
    }
}

File: Halstead_python_example/Calculator.py
<<<<<<< HEAD
class Calculator:
    # adds to positive integers and return the result
    #faulty inputs should return -1
    def add(self, a,b):
        if (a < 0 or b < 0): return -1
        return a+b 
            

    # substract the second from the first param if they are positiv and return the result
    #faulty inputs should return -1
    def sub(self, c,d):
        if (c < 0 or d < 0): return -1

        return c-d
    
    # devide one/two and return the result
    #faulty inputs should return -1
    def divide(self, one,two):
        if two == 0: return -1
        return one/two
    

    # sum the two positive integers and return the result.
    #faulty inputs should return -1
    def sum(self, first,second):
        if(first < 0 or second < 0): return -1
        return first*second
=======

class Calculator:
    # adds to positive integers and return the result
    def add(self, a,b):
        return 0
    
    # substract the second from the first param if they are positiv and return the result
    def sub(self, c,d):
        return 0
    
    # devide one/two and return the result
    def divide(self, one,two):
        return 0
    

    # sum the two positive integers and return the result.
    def sum(self, first,second):
        return 0
>>>>>>> master
